File: backend/app/models/datasource.py
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel


class DataSourceBase(BaseModel):
    type: Literal["prometheus", "influxdb", "loki", "jaeger"]
    name: str
    url: str
    auth_type: Literal["none", "basic", "bearer", "api_key"] = "none"
    auth_config: Optional[dict] = None


class DataSourceCreate(DataSourceBase):
    pass


class DataSourceUpdate(BaseModel):
    name: Optional[str] = None
    url: Optional[str] = None
    auth_type: Optional[str] = None
    auth_config: Optional[dict] = None


class DataSource(DataSourceBase):
    id: str
    status: str
    last_check: Optional[datetime] = None
    error_message: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class DataSourceTestResult(BaseModel):
    success: bool
    message: str
    latency_ms: Optional[int] = None

File: frontend/src/types/datasource.ts
export interface DataSource {
  id: string
  type: 'prometheus' | 'influxdb' | 'loki' | 'jaeger'
  name: string
  url: string
  auth_type: 'none' | 'basic' | 'bearer' | 'api_key'
  auth_config?: {
    username?: string
    password?: string
    token?: string
    api_key?: string
  }
  status: 'connected' | 'disconnected' | 'error' | 'not_configured'
  last_check?: string
  error_message?: string
  created_at: string
  updated_at: string
}

export interface DataSourceCreate {
  type: 'prometheus' | 'influxdb' | 'loki' | 'jaeger'
  name: string
  url: string
  auth_type?: 'none' | 'basic' | 'bearer' | 'api_key'
  auth_config?: Record<string, string>
}

export interface DataSourceUpdate {
  name?: string
  url?: string
  auth_type?: string
  auth_config?: Record<string, string>
}

export interface DataSourceTestResult {
  success: boolean
  message: string
  latency_ms?: number
}
